### ISSUE TITLE
detect question

### ISSUE THREAD

**Issue by trra1988** [ultralytics/yolov5#3254]
## Additional context
hello when I run detect.py the code didn't show any error and the result save in the file, but the picture in file don't have any bbox and class in picture,  the run result is below, thanks

(base) lin@lin:~/yolov5-master$ python detect.py --source data/images --weights yolov5s.pt --conf 0.25
Namespace(agnostic_nms=False, augment=False, classes=None, conf_thres=0.25, device='', exist_ok=False, hide_conf=False, hide_labels=False, img_size=640, iou_thres=0.45, line_thickness=3, name='exp', nosave=False, project='runs/detect', save_conf=False, save_crop=False, save_txt=False, source='data/images', update=False, view_img=False, weights=['yolov5s.pt'])
YOLOv5 🚀 2021-4-27 torch 1.8.1+cu111 CUDA:0 (GeForce GTX 1660 Ti, 5944.625MB)

Fusing layers... 
Model Summary: 224 layers, 7266973 parameters, 0 gradients, 17.0 GFLOPS
image 1/2 /home/lin/yolov5-master/data/images/bus.jpg: 640x480 Done. (0.045s)
image 2/2 /home/lin/yolov5-master/data/images/zidane.jpg: 384x640 Done. (0.041s)
Results saved to runs/detect/exp3
Done. (0.138s)

**Comment by trra1988** [comment]
>  Hello @trra1988, thank you for your interest in  YOLOv5! Please visit our  [Tutorials](https://docs.ultralytics.com/yolov5) to get started, where you can find quickstart guides for simple tasks like [Custom Data Training](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data) all the way to advanced concepts like [Hyperparameter Evolution](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution).
> 
> If this is a  Bug Report, please provide screenshots and **minimum viable code to reproduce your issue**, otherwise we can not help you.
> 
> If this is a custom training  Question, please provide as much information as possible, including dataset images, training logs, screenshots, and a public link to online [W&B logging](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data#visualize) if available.
> 
> For business inquiries or professional support requests please visit https://www.ultralytics.com or email Glenn Jocher at [glenn.jocher@ultralytics.com](mailto:glenn.jocher@ultralytics.com).
> 
> ## Requirements
> Python 3.8 or later with all [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) dependencies installed, including `torch>=1.7`. To install run:
> 
> ```shell
> $ pip install -r requirements.txt
> ```
> 
> ## Environments
> YOLOv5 may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
> 
> * **Google Colab and Kaggle** notebooks with free GPU: [![Open In Colab](https://camo.githubusercontent.com/84f0493939e0c4de4e6dbe113251b4bfb5353e57134ffd9fcab6b8714514d4d1/68747470733a2f2f636f6c61622e72657365617263682e676f6f676c652e636f6d2f6173736574732f636f6c61622d62616467652e737667)](https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb) [![Open In Kaggle](https://camo.githubusercontent.com/a08ca511178e691ace596a95d334f73cf4ce06e83a5c4a5169b8bb68cac27bef/68747470733a2f2f6b6167676c652e636f6d2f7374617469632f696d616765732f6f70656e2d696e2d6b6167676c652e737667)](https://www.kaggle.com/models/ultralytics/yolov5)
> * **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
> * **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
> * **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) [![Docker Pulls](https://camo.githubusercontent.com/280faedaf431e4c0c24fdb30ec00a66d627404e5c4c498210d3f014dd58c2c7e/68747470733a2f2f696d672e736869656c64732e696f2f646f636b65722f70756c6c732f756c7472616c79746963732f796f6c6f76353f6c6f676f3d646f636b6572)](https://hub.docker.com/r/ultralytics/yolov5)
> 
> ## Status
> ![CI CPU testing](https://github.com/ultralytics/yolov5/workflows/CI%20CPU%20testing/badge.svg)
> 
> If this badge is green, all [YOLOv5 GitHub Actions](https://github.com/ultralytics/yolov5/actions) Continuous Integration (CI) tests are currently passing. CI tests verify correct operation of YOLOv5 training ([train.py](https://github.com/ultralytics/yolov5/blob/master/train.py)), testing ([test.py](https://github.com/ultralytics/yolov5/blob/master/test.py)), inference ([detect.py](https://github.com/ultralytics/yolov5/blob/master/detect.py)) and export ([export.py](https://github.com/ultralytics/yolov5/blob/master/models/export.py)) on MacOS, Windows, and Ubuntu every 24 hours and on every commit.

Duplicate of #

>  Hello @trra1988, thank you for your interest in  YOLOv5! Please visit our  [Tutorials](https://docs.ultralytics.com/yolov5) to get started, where you can find quickstart guides for simple tasks like [Custom Data Training](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data) all the way to advanced concepts like [Hyperparameter Evolution](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution).
> 
> If this is a  Bug Report, please provide screenshots and **minimum viable code to reproduce your issue**, otherwise we can not help you.
> 
> If this is a custom training  Question, please provide as much information as possible, including dataset images, training logs, screenshots, and a public link to online [W&B logging](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data#visualize) if available.
> 
> For business inquiries or professional support requests please visit https://www.ultralytics.com or email Glenn Jocher at [glenn.jocher@ultralytics.com](mailto:glenn.jocher@ultralytics.com).
> 
> ## Requirements
> Python 3.8 or later with all [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) dependencies installed, including `torch>=1.7`. To install run:
> 
> ```shell
> $ pip install -r requirements.txt
> ```
> 
> ## Environments
> YOLOv5 may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
> 
> * **Google Colab and Kaggle** notebooks with free GPU: [![Open In Colab](https://camo.githubusercontent.com/84f0493939e0c4de4e6dbe113251b4bfb5353e57134ffd9fcab6b8714514d4d1/68747470733a2f2f636f6c61622e72657365617263682e676f6f676c652e636f6d2f6173736574732f636f6c61622d62616467652e737667)](https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb) [![Open In Kaggle](https://camo.githubusercontent.com/a08ca511178e691ace596a95d334f73cf4ce06e83a5c4a5169b8bb68cac27bef/68747470733a2f2f6b6167676c652e636f6d2f7374617469632f696d616765732f6f70656e2d696e2d6b6167676c652e737667)](https://www.kaggle.com/models/ultralytics/yolov5)
> * **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
> * **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
> * **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) [![Docker Pulls](https://camo.githubusercontent.com/280faedaf431e4c0c24fdb30ec00a66d627404e5c4c498210d3f014dd58c2c7e/68747470733a2f2f696d672e736869656c64732e696f2f646f636b65722f70756c6c732f756c7472616c79746963732f796f6c6f76353f6c6f676f3d646f636b6572)](https://hub.docker.com/r/ultralytics/yolov5)
> 
> ## Status
> ![CI CPU testing](https://github.com/ultralytics/yolov5/workflows/CI%20CPU%20testing/badge.svg)
> 
> If this badge is green, all [YOLOv5 GitHub Actions](https://github.com/ultralytics/yolov5/actions) Continuous Integration (CI) tests are currently passing. CI tests verify correct operation of YOLOv5 training ([train.py](https://github.com/ultralytics/yolov5/blob/master/train.py)), testing ([test.py](https://github.com/ultralytics/yolov5/blob/master/test.py)), inference ([detect.py](https://github.com/ultralytics/yolov5/blob/master/detect.py)) and export ([export.py](https://github.com/ultralytics/yolov5/blob/master/models/export.py)) on MacOS, Windows, and Ubuntu every 24 hours and on every commit.

hello, Thanks for reply  and its help me, but I still can not detect the object, this is my train result and the detect picture
![confusion_matrix](https://user-images.githubusercontent.com/65715677/118945011-a67f0d80-b987-11eb-9e2e-5d61d69284f7.png)
![F1_curve](https://user-images.githubusercontent.com/65715677/118945019-a7b03a80-b987-11eb-9b50-759d40a5ee5f.png)
![labels](https://user-images.githubusercontent.com/65715677/118945023-a8e16780-b987-11eb-8913-dc41a7f7d04f.jpg)
![labels_correlogram](https://user-images.githubusercontent.com/65715677/118945024-a8e16780-b987-11eb-8d31-579ac724291b.jpg)
![P_curve](https://user-images.githubusercontent.com/65715677/118945027-a979fe00-b987-11eb-8702-3fa8694285be.png)
![PR_curve](https://user-images.githubusercontent.com/65715677/118945034-aa129480-b987-11eb-825c-918dd4dde460.png)
![R_curve](https://user-images.githubusercontent.com/65715677/118945037-aaab2b00-b987-11eb-9177-08beab110668.png)
![results](https://user-images.githubusercontent.com/65715677/118945041-ab43c180-b987-11eb-9e0b-2f5da1ee7b1a.png)
[results.txt](https://github.com/ultralytics/yolov5/files/6514100/results.txt)
![test_batch0_labels](https://user-images.githubusercontent.com/65715677/118945047-ad0d8500-b987-11eb-9406-64bf33aeadb5.jpg)
![test_batch0_pred](https://user-images.githubusercontent.com/65715677/118945049-ada61b80-b987-11eb-8e9c-f84d1feccc62.jpg)
![test_batch1_labels](https://user-images.githubusercontent.com/65715677/118945052-ae3eb200-b987-11eb-8f60-380c687d5841.jpg)
![test_batch1_pred](https://user-images.githubusercontent.com/65715677/118945053-aed74880-b987-11eb-8f6d-142669ea6477.jpg)
![test_batch2_labels](https://user-images.githubusercontent.com/65715677/118945058-af6fdf00-b987-11eb-844c-e35cd51a66d8.jpg)
![test_batch2_pred](https://user-images.githubusercontent.com/65715677/118945062-b0087580-b987-11eb-8b1e-a81f262a4038.jpg)
![train_batch0](https://user-images.githubusercontent.com/65715677/118945066-b0a10c00-b987-11eb-8804-681384e41998.jpg)
![train_batch1](https://user-images.githubusercontent.com/65715677/118945072-b139a280-b987-11eb-96a5-cab1b0591962.jpg)
![train_batch2](https://user-images.githubusercontent.com/65715677/118945073-b1d23900-b987-11eb-9c2a-42d1d75e397c.jpg)
thanks for the help.

**Comment by trra1988** [comment]
this is the result of detect
![bus](https://user-images.githubusercontent.com/65715677/118945456-0bd2fe80-b988-11eb-9497-3f46cb5a7463.jpg)

**Comment by 529035872** [comment]
I had the same problem

**Comment by glenn-jocher** [comment]
@529035872 @trra1988 hi thanks for the bug notice! There seems to be an issue with Windows and/or Conda environments that causes detect.py to not detect anything sometimes. I'm not sure of the cause or the solution, so all I can do is point you to one of our verified environments below, where everything will work correctly:


## Environments

YOLOv5 may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Google Colab and Kaggle** notebooks with free GPU: <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>


## Status

![CI CPU testing](https://github.com/ultralytics/yolov5/workflows/CI%20CPU%20testing/badge.svg)

If this badge is green, all [YOLOv5 GitHub Actions](https://github.com/ultralytics/yolov5/actions) Continuous Integration (CI) tests are passing. These tests evaluate proper operation of basic YOLOv5 functionality, including training ([train.py](https://github.com/ultralytics/yolov5/blob/master/train.py)), testing ([test.py](https://github.com/ultralytics/yolov5/blob/master/test.py)), inference ([detect.py](https://github.com/ultralytics/yolov5/blob/master/detect.py)) and export ([export.py](https://github.com/ultralytics/yolov5/blob/master/models/export.py)) on MacOS, Windows, and Ubuntu.

**Comment by 529035872** [comment]
@glenn-jocher  thanks for your reply, I've solve this problem. 
In yolo.py, 
export = False  # onnx export
if export=True, no objects will be detected

**Comment by glenn-jocher** [comment]
@529035872 yolo.py has no `export` variable in the entire file. Your code is out of date. Update your code before anything else.

**Comment by 529035872** [comment]
@glenn-jocher ok, thanks.

**Comment by trra1988** [comment]
@glenn-jocher thanks for help

**Comment by trra1988** [comment]
@529035872 hello, I check my yolo.py, there do not have this line(export = False # onnx export), this cause of you change your own code right?

**Comment by wudashuo** [comment]
If you use **Windows** and there is no detection result at all when using detect.py, then you may meet an environmental problem. You should downgrade your CUDA to 10.2 and reinstall pytorch1.8.1+cu102.
I tested CUDA11.1, 11.2, and 11.3, with PyTorch 1.8.1, none of these worked, only CUDA10.2 worked.

**Comment by trra1988** [comment]
@wudashuo ok, thanks

**Comment by trra1988** [comment]
@wudashuo  hello, thanks for the reply last time, I change my cuda to the vision 11.0 and the code is runing on the ubuntu, but I still can not dectect the object, have you run on ubuntu before? thanks

**Comment by wudashuo** [comment]
> @wudashuo hello, thanks for the reply last time, I change my cuda to the vision 11.0 and the code is runing on the ubuntu, but I still can not dectect the object, have you run on ubuntu before? thanks

I met several environmental problems on Windows, but I never met a problem on Ubuntu, no matter CUDA10.2, CUDA11.1, or CUDA11.2. Have you changed the codes of plotting?

**Comment by trra1988** [comment]
> > @wudashuo hello, thanks for the reply last time, I change my cuda to the vision 11.0 and the code is runing on the ubuntu, but I still can not dectect the object, have you run on ubuntu before? thanks
> 
> I met several environmental problems on Windows, but I never met a problem on Ubuntu, no matter CUDA10.2, CUDA11.1, or CUDA11.2. Have you changed the codes of plotting?

No ,I did not, I just detect with pretrain model

**Comment by trra1988** [comment]
@wudashuo The problem is solved, I change my system to ubuntu18.04 and change cuda to 10.2, The problem is cause by vision of cuda, Thanks

**Comment by trra1988** [comment]
@wudashuo The problem is solved, I changed my system to ubuntu18.04 and change cuda to 10.2, the problem is cause by vision of cuda. Thanks

**Comment by glenn-jocher** [comment]
@wudashuo @trra1988 another user found that FP32 inference worked for them but not FP16 inference (the default in detect.py). 

Can you see if the problem also happens on your system with YOLOv5 PyTorch Hub models? The Hub models are loaded as FP32 and use a different inference pathway (with AMP) which does not convert them to FP16. Try this line:
```
python hubconf.py
```

You should see
```
Fusing layers... 
Model Summary: 224 layers, 7266973 parameters, 0 gradients
Adding AutoShape... 
YOLOv5 🚀 v5.0-112-g9f3a388 torch 1.8.1 CPU

image 1/5: 720x1280 2 persons, 2 ties
image 2/5: 720x1280 2 persons, 2 ties
image 3/5: 1080x810 4 persons, 1 bus
image 4/5: 1080x810 4 persons, 1 bus
image 5/5: 320x640
Speed: 100.4ms pre-process, 249.1ms inference, 1.7ms NMS per image at shape (5, 3, 640, 640)
Saved zidane.jpg, zidane.jpg, image2.jpg, bus.jpg, image4.jpg to runs/hub/exp5
```
